File: 6-kyu/name-to-matrix/python/solution.py
from math import ceil, sqrt

def matrixfy(st):
    if len(st) == 0:
        return "name must be at least one letter"
    idx = 0
    length = ceil(sqrt(len(st)))
    ret = []
    for x in range(length):
        row = []
        for y in range(length):
            row.append(st[idx] if idx < len(st) else '.')
            idx += 1
        ret.append(row)
    return ret

File: 6-kyu/name-to-matrix/javascript/solution.js
function matrixfy(str) {
  if (str.length === 0) {
    return 'name must be at least one letter';
  }
  const nextBiggestSquare = Math.ceil(Math.sqrt(str.length)) ** 2;
  const padded = str.padEnd(nextBiggestSquare, '.').split("");
  const matrix = [];
  const rowLength = Math.sqrt(nextBiggestSquare);
  for (let i = 0; i < rowLength; i++) {
    const row = [];
    for (let j = 0; j < rowLength; j++) {
      const next = padded.shift();
      row.push(next);
    }
    matrix.push(row);
  }
  return matrix;
}